File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/PublishableExceptions/SeatInsufficientException.cs
using System;
using System.Collections.Generic;
using ENode.Infrastructure;

namespace ConferenceManagement
{
    public class SeatInsufficientException : PublishableException
    {
        public Guid ConferenceId { get; private set; }
        public Guid ReservationId { get; private set; }

        public SeatInsufficientException(Guid conferenceId, Guid reservationId) : base()
        {
            ConferenceId = conferenceId;
            ReservationId = reservationId;
        }

        public override void RestoreFrom(IDictionary<string, string> serializableInfo)
        {
            ConferenceId = Guid.Parse(serializableInfo["ConferenceId"]);
            ReservationId = Guid.Parse(serializableInfo["ReservationId"]);
        }
        public override void SerializeTo(IDictionary<string, string> serializableInfo)
        {
            serializableInfo.Add("ConferenceId", ConferenceId.ToString());
            serializableInfo.Add("ReservationId", ReservationId.ToString());
        }
    }
}

File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/publishableexceptions/SeatInsufficientException.java
package com.microsoft.conference.management.domain.publishableexceptions;

import org.enodeframework.domain.DomainException;

import java.util.Map;

public class SeatInsufficientException extends DomainException {
    public String ConferenceId;
    public String ReservationId;

    public SeatInsufficientException(String conferenceId, String reservationId) {
        super();
        ConferenceId = conferenceId;
        ReservationId = reservationId;
    }

    @Override
    public void serializeTo(Map<String, Object> serializableInfo) {
        ConferenceId = (String) serializableInfo.get("ConferenceId");
        ReservationId = (String) serializableInfo.get("ReservationId");
    }

    @Override
    public void restoreFrom(Map<String, Object> serializableInfo) {
        serializableInfo.put("ConferenceId", ConferenceId);
        serializableInfo.put("ReservationId", ReservationId);
    }
}
